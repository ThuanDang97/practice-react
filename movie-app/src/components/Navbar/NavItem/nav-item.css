.nav-link {
  padding: 5px 20px;
  text-align: center;
}

.nav-link:hover {
  background-color: #ced7df;
}
.nav-link span {
  color: black;
  text-decoration: none;
}
