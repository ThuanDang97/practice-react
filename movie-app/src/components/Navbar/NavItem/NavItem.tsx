import { Link } from 'react-router-dom';
import { IItem } from '../../../interfaces/Navbar/IItem';
import './nav-item.css';
type TProps = {
  itemNav: IItem[];
};

const NavItem = (props: TProps): JSX.Element => {
  const { itemNav } = props;

  return (
    <div className="dropdown-content">
      {itemNav.map((item) => (
        <Link to={item.pathItem} key={item.idItem} className="nav-link">
          <span>{item.titleItem}</span>
        </Link>
      ))}
    </div>
  );
};

export { NavItem };
