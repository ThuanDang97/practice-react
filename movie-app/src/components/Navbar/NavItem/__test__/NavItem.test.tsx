import { render, screen } from '@testing-library/react';
import { NavItem } from '../NavItem';
import { BrowserRouter as Router } from 'react-router-dom';
describe('Test Nav Item component', () => {
  const item = [{ idItem: 1, titleItem: 'Popular', pathItem: '/' }];

  const component = (
    <Router>
      <NavItem itemNav={item} />
    </Router>
  );

  test('Component Nav Item matches DOM Snapshot', () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Nav Item should render correctly', () => {
    render(component);

    expect(screen.getByTestId('navitem')).toHaveTextContent('Popular');
  });
});
