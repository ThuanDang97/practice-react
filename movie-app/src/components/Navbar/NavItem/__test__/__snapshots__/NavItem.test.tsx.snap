// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Nav Item component Component Nav Item matches DOM Snapshot 1`] = `
<div>
  <div
    class="dropdown-content"
  >
    <a
      class="nav-link"
      data-testid="navitem"
      href="/"
    >
      <span>
        Popular
      </span>
    </a>
  </div>
</div>
`;
